File: src/app/model/danhgia.ts
export class dg {
    _id: number;
    sosao: number;
    ten: string;
    sdt:string;
    email:string;
    noidung:string;
    hinh:Array<string>;
    luotthich:number;
    danhgiaphu:Array<dgphu>;
    tieuchidanhgia:Array<boolean>;
    _id_sanpham:number;
    ngaydanhgia:Date;
    kiemduyet:boolean;
    constructor(_id: number,  sosao: number, ten: string, sdt:string, email:string, noidung:string, hinh:Array<string>, luotthich:number, danhgiaphu:Array<dgphu>, tieuchidanhgia:Array<boolean>, _id_sanpham:number,  kiemduyet:boolean) {
        this._id = _id;
        this.sosao = sosao;
        this.ten = ten;
        this.sdt =sdt;
        this.email = email;
        this.noidung = noidung;
        this.hinh = hinh;
        this.luotthich = luotthich;
        this.danhgiaphu = danhgiaphu;
        this.tieuchidanhgia = tieuchidanhgia;
        this._id_sanpham = _id_sanpham;
        this.kiemduyet = kiemduyet;
    }
}

export class dgphu {
    noidung:string;
    luotthich:number;
    ten: string;
    chucdanh : boolean;
    gioitinh: boolean;
    email:string;
    ngaydanhgiaphu:Date;
    kiemduyetphu:boolean;
    constructor(noidung:string, luotthich:number, ten: string, chucdanh : boolean, gioitinh: boolean, email:string, kiemduyetphu:boolean) {
        this.noidung = noidung;
        this.luotthich = luotthich;
        this.ten = ten;
        this.chucdanh = chucdanh;
        this.gioitinh = gioitinh;
        this.email = email;
        this.kiemduyetphu = kiemduyetphu;
    }
}

File: dienmayxanhapi/dienmayxanhapi/Model/danhgia.cs
using MongoDB.Bson.Serialization.Attributes;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace dienmayxanhapi.Model
{
  public class danhgia
  {
    public int? _id { get; set; }
    public int? sosao { get; set; }
    public string ten { get; set; }
    public string sdt { get; set; }
    public string email { get; set; }
    public string noidung { get; set; }
    public List<string> hinh { get; set; }
    public int? luotthich { get; set; }
    public IList<danhgiaphu> danhgiaphu { get; set; }
    public List<Boolean> tieuchidanhgia { get; set; }
    public int? _id_sanpham { get; set; }
    [BsonDateTimeOptions(Kind = DateTimeKind.Utc)]
    public DateTime ngaydanhgia { get; set; }

  }

  public class danhgiaphu
  {
    public string noidung { get; set; }
    public int? luotthich { get; set; }
    public string ten { get; set; }
    public Boolean gioitinh { get; set; }
    public string email { get; set; }
    public DateTime ngaydanhgiaphu { get; set; }
  }
}
